common/src/androidTest/kotlin/JSONArrayBugInstrumentedTest.kt

```diff
@@ -0,0 +1,33 @@
+import android.os.Build
+import com.revenuecat.purchases.common.networking.MapConverter
+import org.json.JSONObject
+import org.junit.Assert.assertEquals
+
+//@org.junit.runner.RunWith(androidx.test.ext.junit.runners.AndroidJUnit4::class)
+class JSONArrayBugInstrumentedTest {
+    @org.junit.Test
+    fun testReproduceIncorrectJSONArray() {
+        // these tests should only be run on Android 4, since that's the version that has the bug.
```

Comment:
This is going to be difficult to automate since Firebase test labs doesn't support these versions... I guess we could create our own emulator image and upload it somewhere so it's accesible through CI, but it's going to be tricky to setup...